tmt/utils/__init__.py

```diff
@@ -4277,54 +4277,84 @@ def fmf_id(
         *,
         name: str,
         fmf_root: Path,
-        always_get_ref: bool = False,
         logger: tmt.log.Logger) -> 'tmt.base.FmfId':
     """ Return full fmf identifier of the node """
 
     def run(command: Command) -> str:
-        """ Run command, return output """
-        try:
-            result = command.run(cwd=fmf_root, logger=logger)
-            if result.stdout is None:
-                return ""
-            return result.stdout.strip()
-        except RunError:
-            # Always return an empty string in case 'git' command is run in a non-git repo
+        """ Run command, return output. We don't need the stderr here, but we need exit status. """
+        result = command.run(cwd=fmf_root, logger=logger)
+        if result.stdout is None:
             return ""
+        return result.stdout.strip()
 
     from tmt.base import FmfId
 
     fmf_id = FmfId(fmf_root=fmf_root, name=name)
 
     # Prepare url (for now handle just the most common schemas)
-    branch = run(Command("git", "rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{u}"))
     try:
-        remote_name = branch[:branch.index('/')]
-    except ValueError:
-        remote_name = 'origin'
-    remote = run(Command("git", "config", "--get", f"remote.{remote_name}.url"))
-    fmf_id.url = public_git_url(remote) if remote else None
-
-    # Construct path (if different from git root)
-    fmf_id.git_root = git_root(fmf_root=fmf_root, logger=logger)
-
-    if fmf_id.git_root:
+        # Check if we are a git repo
+        run(Command("git", "rev-parse", "--is-inside-work-tree"))
+
+        # Initialize common git facts
+        # Get some basic git references for HEAD
+        all_refs = run(Command("git", "for-each-ref", "--format=%(refname)", "--points-at=@"))
+        logger.debug(f"git all_refs: {all_refs}")
```

Comment:
It seems that every call should be rewritten to `debug(key, value)` form, as all calls do log some label and a corresponding value: `logger.debug('git all_refs', all_refs)`

Reply:
Argh I did not realize the API is completely different from `logging.Logger`. What is the `key`-`value` supposed to represent in this case?

Reply:
It's not like a key/value in mapping, but rather a named, labeled value, exactly what you have here: "git root" + "/foo/bar/baz", "git ref" + "deadbeef", etc.

Reply:
Ok, updated and rebased. Had to navigate a bit deeper into the implementation to understand more on how it works, basically `key` is like a one-line `summary` and `value` is the formatted `detail` that are split-lined and indented. Thoughts on renaming these arguments to `summary` and `details`?

Also got me thinking if these should be in a topic, or are there tips of when to use/not use topics?

Reply:
Good question: we don't have any tips WRT adding new topics. I guess the general idea is to use them for logging things that are easy to reproduce when needed, i.e. derived from CLI, fmf files, or other reproducible conditions. `adjust` decisions, CLI/fmf normalization, help rendering. If a user hits a problem, we can re-run tmt with the given input and enabled topics, and reproduce the behavior. And it would be too much output when logging by default.

These git operations do fit the description, at least from my limited understanding - if I get it right, if the user has trouble with what tmt inferred from git repo, it's always possible to feed it with the original URL, ref, path, etc. and repeat the run. On the other hand, there is no rule that everything that matches the description above must be moved behind a logging topic - I believe some tmt developers and users do like to see some types of info in their logs by default. But it's just my observation, we do not have anything codified in our contribution docs - can be done, start a discussion and let's see what community thinks about it :)

Reply:
Is it possible to have multiple topics for a message. I think having a common topic of `git` would be helpful, and then subtopics, `git_metadata`, `fmf_metadata`, etc.

Why I was thinking about this is that the debug log for hitting `fmf_id` would have been huge (less so now that I've moved the git part to be cached) so it might be good to make these topics a bit more flexible

Reply:
Hmm, it's not possible to have multiple topics, but the internal API shouldn't be hard to extend. The use case makes sense to me.